### PR TITLE
Using jsondiff package to print itests diffs and skipping random stop IDs

### PR DESCRIPTION
**Pull request by lboyarsky**
Ticket: [OTP integration tests: use JSON diff to better see issues](https://app.asana.com/0/810933294009540/1134882691756336/f)

Prints diffs as following:

```bash
Comparing plans from Wellington::42.40237,-71.077082 to 49 Cushing Street, Medford, MA, USA::42.4142089,-71.0968295 departing on 2019-09-04 08:20:00
[FAIL] Plans are different:
{2: {'legs': {1: {'to': {'stopId': '1:temp_0.07696220697550493'}}, 2: {'from': {'stopId': '1:temp_0.07696220697550493'}}}}}
```

(says stop ids are different).

### CODE REVIEW

**Comment by paulswartz** [comment]
I don't understand the failure in this last build: https://semaphoreci.com/mbta/otp-deploy/servers/otp-dev/deploys/490

**Comment by lboyarsky** [comment]
This failure implies something like that --
image

Basically the error message says that the values of `stopId` field in `from` and `to` sections are different in the first json from the values in the same fields in the second json. 

1) I'll see how to print this in more informative way. 
2) I'm also fixing the underlying cause for this failure, which is temporary stops introduced in https://github.com/mbta/OpenTripPlanner/pull/21 having random IDs each time, I'm going to remove them from comparison.

**Comment by lboyarsky** [comment]
Now the diff from this -- 

image

is printed like: 

```bash
Comparing plans from Wellington::42.40237,-71.077082 to 49 Cushing Street, Medford, MA, USA::42.4142089,-71.0968295 arriving by 2019-09-04 18:46:00
[FAIL] Plans are different:
{
  "1": {
    "legs": {
      "1": {
        "to": {
          "stopId": [
            "1:temp_0.36860661842879305",
            "1:temp_0.649375821136542"
          ]
        }
      },
      "2": {
        "from": {
          "stopId": [
            "1:temp_0.36860661842879305",
            "1:temp_0.649375821136542"
          ]
        }
      }
    }
  }
}
```

Which, I think, is more clear.

**Comment by lboyarsky** [comment]
I've added the fix for the random stop IDs here as well.

File: semaphore/integration_tests.py [hunk 1]
```diff
@@ -87,37 +89,63 @@ def get_trip_plans(environment, from_place, to_place, trip_date, arrive_by):
...
+        if isinstance(json, list):
+            return [remove_random_stop_ids(j) for j in json]
+        elif isinstance(json, dict):
+            result = {}
+            for key, value in json.items():
+                if key == "stopId":
+                    result[key] = re.sub(r"(\w+:temp_0)\.\d+", r"\1-random", value)
+                else:
+                    result[key] = value
+            return result
```
Comment (paulswartz) [inline semaphore/integration_tests.py]: I think you can simplify this:
```python
if isinstance(json, list):
  return [remove_random_stop_ids(j) for j in json]

if isinstance(json, dict) and "stopId" in json:
  json = json.copy()
  json["stopId"] = re.sub(...)

return json
```

Comment (lboyarsky) [reply semaphore/integration_tests.py]: I don't think it will process nested dictionaries correctly like that.

Comment (paulswartz) [reply semaphore/integration_tests.py]: You mean where `value` is a dictionary? I don't see that happening here, ether.

Comment (lboyarsky) [reply semaphore/integration_tests.py]: Yep, I've fixed it in the latest commit.

**Pull request merged**